### PR TITLE
Added new endpoint to search for existing bceid for Volunteers

### PR DESCRIPTION
**Pull request by CodingVelocista**

### CODE REVIEW

File: embc-app/Controllers/VolunteersController.cs [hunk 1]
```diff
@@ -47,10 +47,25 @@ public async Task<IActionResult> Get(string id)
...
+        public async Task<IActionResult> BceIdExists(string bceid)
+        {
+            if (string.IsNullOrWhiteSpace(bceid))
+            {
+                return BadRequest(Json(bceid));
+            }
+            var existing = await dataInterface.GetVolunteerByExternalIdAsync(bceid);
+            if (existing != null)
+            {
+                return BadRequest();
```
Comment (ytqsl) [inline embc-app/Controllers/VolunteersController.cs]: This doesn't make sense, shouldn't it return boolean instead?

Comment (CodingVelocista) [reply embc-app/Controllers/VolunteersController.cs]: This is how Alejandro wanted it.

If it returns a badrequest then the BCEID is safe to use if OK then a duplicate was found.

Comment (ytqsl) [reply embc-app/Controllers/VolunteersController.cs]: @asanchezr , comment?

Comment (asanchezr) [reply embc-app/Controllers/VolunteersController.cs]: Hey, what I suggested is to do something like this - https://softwareengineering.stackexchange.com/questions/347148/restful-service-for-checking-if-email-is-unique

I like the suggestion of doing the following:

GET /api/volunteers/bceid/xxxyyyzzz123

* Return 200 OK if bceid found
* Return 404 NOT FOUND if bceid not found

Comment (ytqsl) [reply embc-app/Controllers/VolunteersController.cs]: @CodingVelocista 400!=404

Comment (asanchezr) [reply embc-app/Controllers/VolunteersController.cs]: The above is generic enough and the front-end would translate the HTTP status to a validation message. So no need for a json response. Now the logic above is "inverted". What I would expect is:

```
if (existing != null) {
  return Ok();
}
return NotFound();
```

Comment (CodingVelocista) [reply embc-app/Controllers/VolunteersController.cs]: I knew that!

**Pull request merged**